kale/embed/mpca.py

```diff
@@ -182,15 +183,16 @@ def _fit(self, x):
                     xj_unfold = unfold(xj, i - 1)
                     phi[i] = np.dot(xj_unfold, xj_unfold.T) + phi[i]
 
-                eig_values, eig_vectors = np.linalg.eig(phi[i])
+                # eig_values, eig_vectors = np.linalg.eig(phi[i])
+                eig_vectors, eig_values, vt = np.linalg.svd(phi[i])
                 idx_sorted = (-1 * eig_values).argsort()
                 proj_mats[i - 1] = (eig_vectors[:, idx_sorted][:, : shape_out[i - 1]]).T
 
         x_projected = multi_mode_dot(x, proj_mats,
                                      modes=[m for m in range(1, n_dims)])
         x_proj_unfold = unfold(x_projected, mode=0)  # unfold the tensor projection to shape (n_samples, n_features)
         # x_proj_cov = np.diag(np.dot(x_proj_unfold.T, x_proj_unfold))  # covariance of unfolded features
-        x_proj_cov = np.sum(np.multiply(x_proj_unfold, x_proj_unfold), axis=1)  # memory saving computing covariance
+        x_proj_cov = np.sum(np.multiply(x_proj_unfold.T, x_proj_unfold.T), axis=1)  # memory saving computing covariance
```

Comment:
`.T` means the transpose, right? Is it wrong previously? We should have some testing implemented to verify the correctness of code. I understand that we are at level zero but MPCA maybe a good starting point since I/we know best how it should behave.

Reply:
It was wrong. Previously we take the diagnoal of covariance matrix (commented above). It was changed to `np.multiply` in the last update because Cameron reported a memory issue of `np.dot`. Then Lawrence pointed this out and I corrected in this PR. I agree to have a test for mpca. I will implement a demo test.

Reply:
This type of problem is identifiable automatically through testing so I do think it is time for us to consider it.